### ISSUE TITLE
Spell-suit thoughts

### ISSUE THREAD

**Issue by Caldoric** [Vazkii/Psi#737]
First off, less an "issues and more of a suggestion/idea.

TL:DR, modular suit upgrades, including spell slots, psi energy caps, and (hopefully) optional cosmetics, in the spirit of how CADs have progression and customization.

In all honesty, from what I've seen of the suits (which, admittedly, isn't much,) you kind of just build the thing out of psimetal and a couple other bits, and then it's stuck at a specific setup forevermore. What I'm suggesting here is the ability to treat the armor itself as a basic framework to be expanded upon, such that you can throw on various additional pieces made of the various materials used in the mod, all for a mix of functional and cosmetic gain. Some options would give range, others power, others extra spell slots, or casting speed, etc. Some would make the armor look sleeker, others would leave it looking more regal, and yet others would lend it a far more aggressive and/or intimidating look. Not to mention, if the colorizer items aren't already usable to change light colors and stuff, then having that as a post-build option would be fun, too.

After all, a good portion of battle is in the mind of the combatants, isn't it?
